### PR TITLE
parser: add NULL DEFINED BY and FORMAT to LOAD DATA

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref #40499

Problem Summary:

add parser support for `NULL DEFINED BY`

```
LOAD DATA ... [LINES ...] [NULL DEFINED BY <string> [OPTIONALLY ENCLOSED]] [IGNORE ...]
```

add parser support for `FORMAT`

```
LOAD DATA ... INFILE 'string' FORMAT 'string' ...
```

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [x] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by lance6716** [comment]
ptal @gozssky @buchuitoudegou

**Review by hawkingrei — APPROVED** [review]

File: parser/parser.y [hunk 1]
```diff
@@ -13970,6 +14013,30 @@ LinesTerminated:
 		$$ = $3
 	}
 
+NullDef:
```
Comment (gozssky) [inline parser/parser.y:14016]: Why not resuse `TextString`?

Comment (lance6716) [reply parser/parser.y]: https://github.com/pingcap/tidb/pull/41332/commits/3326462d9e2eddc50cd78258a8c699de6ba8d4d8

**Review by gozssky — APPROVED** [review]

**Pull request merged**